src/merklize.rs

```diff
@@ -120,32 +125,69 @@ impl Iterator for MerklizedChunks {
 impl MerklizedChunks {
 	/// Compute `MerklizedChunks` from a list of erasure chunks.
 	pub fn compute(chunks: Vec<Vec<u8>>) -> Self {
-		let mut hashes: Vec<Hash> = chunks
-			.iter()
-			.map(|chunk| {
+		let chunks_len = chunks.len();
+		let target_size = chunks_len.next_power_of_two();
+		
+		// Parallel chunk hashing
+		#[cfg(feature = "parallel")]
+		let hashes: Vec<Hash> = {
+			let mut h = chunks
+				.par_iter()
+				.map(|chunk| Hash::from(hash_fn(chunk)))
+				.collect::<Vec<_>>();
+			h.resize(target_size, Hash::default());
+			h
+		};
+		
+		#[cfg(not(feature = "parallel"))]
+		let hashes = {
+			let mut h = Vec::with_capacity(target_size);
+			for chunk in chunks.iter() {
```

Comment:
May simplify changes by replacing `.collect()` with `.collect_with_capacity()`
```rust
trait CollectWithCapacity<T> : Iterator<Item = T> {
    fn collect_with_capacity(self) -> Vec<T>;
}
impl<I: Iterator> CollectWithCapacity<I::Item> for I {
    fn collect_with_capacity(self) -> Vec<I::Item> {
        let mut result = Vec::with_capacity(self.size_hint().0);
        for x in self {
            result.push(x);
        }
        result
    }
}
```

Reply:
`size_hint().0` is equal `chunks_len`. It can be less than `target_size`. Which is cuase additional rellocation in a case where `target_size > chunks_len`.